cmd/soroban-cli/src/config/sign_with.rs

```diff
@@ -66,6 +67,29 @@ impl Args {
         locator: &locator::Args,
         network: &Network,
         quiet: bool,
+    ) -> Result<TransactionEnvelope, Error> {
+        self.sign_env(tx, locator, network, quiet, None).await
+    }
+
+    pub async fn sign_tx_env_with_default_signer(
+        &self,
+        tx: &TransactionEnvelope,
+        locator: &locator::Args,
+        network: &Network,
+        quiet: bool,
+        source_account: UnresolvedMuxedAccount,
+    ) -> Result<TransactionEnvelope, Error> {
+        self.sign_env(tx, locator, network, quiet, Some(source_account))
+            .await
+    }
+
+    async fn sign_env(
+        &self,
+        tx: &TransactionEnvelope,
+        locator: &locator::Args,
+        network: &Network,
+        quiet: bool,
+        source_account: Option<UnresolvedMuxedAccount>,
```

Comment:
What does the `_with_default_signer` mean? Does it mean that the signing will occur with the source account? It looks to me like that is what is happening.

I think it might be easier to follow to remove the first function, and add a comment to `sign_env` (maybe renamed to `sign_tx`), that indicates if a `source_account` is provided, it will sign with it if no other signer is specified. That is what `sign_env` does anyway.